bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/graphics/Image.java

```diff
@@ -2298,6 +2407,38 @@ protected ElementAtZoom<ImageData> loadImageData(int zoom) {
 		return imageDataAtZoom;
 	}
 
+	@Override
+	protected ImageData loadImageData(int targetWidth, int targetHeight) {
+		Optional<String> fileForTargetSize = validateAndGetImagePathAtTargetSize(provider, targetWidth, targetHeight);
+		if(fileForTargetSize.isPresent()) {
+			ImageData imageDataAtZoom = ImageDataLoader.loadByTargetSize(fileForTargetSize.get(), targetWidth, targetWidth);
+			return imageDataAtZoom;
+		}
+		return null;
+	}
+
+	private Optional<String> validateAndGetImagePathAtTargetSize(ImageFileNameProvider provider, int targetWidth, int targetHeight) {
+		if (provider == null) {
+			SWT.error(SWT.ERROR_NULL_ARGUMENT);
+		}
+
+		if (provider instanceof SizeAwareImageFileNameProvider sizeAwareProvider) {
+			Optional<String> imagePathAtTargetSize = getElementAtTargetSize(sizeAwareProvider::getImagePath,
+					targetWidth, targetHeight);
+
+			if (imagePathAtTargetSize.isEmpty()) {
+				SWT.error(SWT.ERROR_INVALID_ARGUMENT, null,
+						": ImageFileNameProvider [" + provider + "] returns no filename at 100% zoom.");
```

Comment:
The error message seems to be wrong, as it's not about returning file names for zooms.